sound/soc/sof/pm.c

```diff
@@ -157,7 +157,7 @@ static int sof_resume(struct device *dev, bool runtime_resume)
 	}
 
 	/* restore pipelines */
-	ret = sof_restore_pipelines(sdev->dev);
+	ret = sof_set_up_pipelines(sdev->dev);
```

Comment:
Even though there's the merit of verifying topology by setting up the widgets during topology load, I removed it in this version because this will make the IPC abstraction work easier. If there are issues with one pipeline in tplg, this will at least allow the others to work. But I have no specific preference. If you want to do the check, I can drop this patch.

Reply:
@ranj063 I am not a big fan either of validating the topology on first boot, and not later.

maybe we could have a kernel parameter to treat all dynamic pipelines as static, that would force a one-shot validation of the topology if needed?